test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -1,2451 +0,0 @@
-						content: 'heading default content',
-						metadata: {
-							bindings: {
-								content: {
-									source: 'core/post-meta',
-									args: { key: 'text_custom_field' },
-								},
-							},
-						},
-					},
-				} );
-
-				// Select the heading and press Enter at the end of it.
-				const heading = editor.canvas.getByRole( 'document', {
-					name: 'Block: Heading',
-				} );
-				await editor.selectBlocks( heading );
-				await page.keyboard.press( 'End' );
-				await page.keyboard.press( 'Enter' );
-				// Can't use `editor.getBlocks` because it doesn't return the meta value shown in the editor.
-				const [ initialHeading, newEmptyParagraph ] =
-					await editor.canvas.locator( '[data-block]' ).all();
-				// First block should be the original block.
-				await expect( initialHeading ).toHaveAttribute(
-					'data-type',
-					'core/heading'
-				);
-				await expect( initialHeading ).toHaveText(
-					'Value of the text custom field'
-				);
-				// Second block should be an empty paragraph block.
-				await expect( newEmptyParagraph ).toHaveAttribute(
-					'data-type',
-					'core/paragraph'
-				);
-				await expect( newEmptyParagraph ).toHaveText( '' );
-				await expect( newEmptyParagraph ).toBeEditable();
-			} );
-			test( 'should show a selector for content', async ( {
-				editor,
-				page,
-			} ) => {
-				await editor.insertBlock( {
-					name: 'core/heading',
-				} );
-				await page.getByLabel( 'Attributes options' ).click();
-				const contentAttribute = page.getByRole( 'menuitemcheckbox', {
-					name: 'Show content',
-				} );
-				await expect( contentAttribute ).toBeVisible();
-			} );
-		} );
-
-		test.describe( 'Button', () => {
-			test( 'should show the value of the custom field when text is bound', async ( {
-				editor,
-			} ) => {
-				await editor.insertBlock( {
-					name: 'core/buttons',
-					innerBlocks: [
-						{
-							name: 'core/button',
-							attributes: {
-								anchor: 'button-text-binding',
-								text: 'button default text',
-								url: '#default-url',
-								metadata: {
-									bindings: {
-										text: {
-											source: 'core/post-meta',
-											args: { key: 'text_custom_field' },
-										},
-									},
-								},
-							},
-						},
-					],
-				} );
-				const buttonBlock = editor.canvas
-					.getByRole( 'document', {
-						name: 'Block: Button',
-						exact: true,
-					} )
-					.getByRole( 'textbox' );
-				await buttonBlock.click();
-				await expect( buttonBlock ).toHaveText(
-					'Value of the text custom field'
-				);
-
-				// Check the frontend shows the value of the custom field.
-				const previewPage = await editor.openPreviewPage();
-				const buttonDom = previewPage.locator(
-					'#button-text-binding a'
-				);
-				await expect( buttonDom ).toHaveText(
-					'Value of the text custom field'
-				);
-				await expect( buttonDom ).toHaveAttribute(
-					'href',
-					'#default-url'
-				);
-			} );
```

Comment:
Unify testing all button attributes here for custom sources: https://github.com/WordPress/gutenberg/pull/65526/files#diff-70af8def41b67ce618e94f6b4f68044bb4540816a1a8084a1ba1eac9e0c26595R106-R140

It lacks checking the frontend.

Post meta is only checked for paragraphs.